docs/rules/no-reference-type-as-default-prop.md

```diff
@@ -0,0 +1,68 @@
+# Prevent usage of referential-type variables as default param in functional component (react/no-reference-type-as-default-prop)
+
+Warns if in a functional component, a referential type value (such as array/object literal) is used as default prop, to prevent potential un-necessary re-renders, and performance regressions.
```

Comment:
i'm not sure if the name "referential type" is helpful - the hazard is with any object, a category which arrays and functions belong to, so `no-object-as-default-prop` might be clearer? (especially since any implication that JS is a pass-by-reference language would be incorrect and confusing)

Reply:
Sounds good. If we see `object` as a broad term, the indication does seem clearer with the proposed name `no-object-as-default-prop`.